### DISCUSSION TITLE
Real world units and springs?

### DISCUSSION THREAD

**Discussion by salk31** [dyn4j/dyn4j#222] [Q&A] [answered]
Hi,

I was initially just going to use dyn4j for the constraint code (totally beyond me) and use my ancient school maths to do the vectors to work out turning forces etc between a set of levers and springs.

But I noticed UnitConversion class and mentions of MKS so am I missing a trick and dyn4j does physics closer to the real world than I thought?

I'm trying to simulate various constant force spring mechanisms.

Is there some obvious way to model a spring I'm missing (in N per m) and then ask the model what forces it is experiencing?

Many thanks for the code. I tried a JS equivalent but I missed all the Java/Eclipse developer toys too much.

Cheers

Sam

**Comment by wnbittle** [comment, marked as answer]
Yes, all units in dyn4j are MKS, shape sizes, velocity, torque, mass, etc. and dyn4j attempts to be "real world" (as close as you can in 2D).  For springs you use the setFrequency method to indirectly set the spring stiffness.  The stiffness is computed based on the frequency and mass of the combined objects.  The thought processing being, instead of specifying an arbitrary constant (k), you specify an oscillation frequency.

You can get the _total_ force applied by a `Joint` (in a single timestep) by using the `getReactionForce(invdt)` method, but it's not specific to the spring only.  The reaction force is the combination of all features of the joint like motors, joints, and the constraint itself.
